src/poll/poll.q.consumer.service.ts

```diff
@@ -66,14 +68,22 @@ export class PollCreateConsumer {
 
         if (poll.strategy_config) {
             // sanitizing block heights
-            let block = null;
-            for await (const strategy of poll.strategy_config) {
-                if (strategy.block_height <= 0) {
-                    if (!block) block = await web3Utils.getEthersProvider(1).getBlockNumber();
-                    strategy.block_height = block - strategy.block_height;
-                }
+            const strategy = poll.strategy_config[0];
+            let mainnetBlock = strategy.block_height.find(BlockHeight => BlockHeight.chain_id === '1').block;
+
+            if (mainnetBlock <= 0) {
+                const block = await web3Utils.getEthersProvider(1).getBlockNumber();
+                mainnetBlock = block - mainnetBlock;
+                strategy.block_height.find(BlockHeight => BlockHeight.chain_id === '1').block = mainnetBlock;
             }
+            
+            // add just polygon for now.
+            const polygonBlock = await this.gqlService.getEquivalentBlock(mainnetBlock, '137');
```

Comment:
What if FE passes polygon blockheight? We could skip this check.

Reply:
You mean just pass current block for now? Yes we could do that. Could also do that here until we implement block height on FE.

Reply:
fixed to return current block if request fails